src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -125,6 +125,12 @@ public static void ThrowInvalidOperationException_SerializerPropertyNameNull(Typ
             throw new InvalidOperationException(SR.Format(SR.SerializerPropertyNameNull, parentType, jsonPropertyInfo.PropertyInfo.Name));
         }
 
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
NoInlining is not necessary on throw helpers like this. It actually degrades performance because it prevents the JIT from realizing that the throw helper never returns.

Reply:
cc @steveharter, @ahsonkhan on the other helpers in this file.